protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```
@@ -258,7 +258,16 @@ impl ChannelFactory {
                 .safe_lock(|ids| ids.new_channel_id(extended_channels_group))
                 .unwrap();
             self.channel_to_group_id.insert(channel_id, 0);
-            let target = crate::utils::hash_rate_to_target(hash_rate, self.share_per_min);
+            let target = match crate::utils::hash_rate_to_target(hash_rate, self.share_per_min) {
```

function should return an error not an option

---

*result